feat(cmd): pipe buffer to &shell with :! (#23720)
neovim/neovim: runtime/doc/change.txt

							*!*
!{motion}{filter}	Filter {motion} text lines through the external

> 			program {filter}. If {filter} is empty, use previously
> 			{filter} program. {filter} is initially set to 'shell'
> 			by default on startup.

Reviewer: ```suggestion
			program {filter}. If {filter} is empty, use previous
			{filter} program. Defaults to 'shell'.
```

similar for the other case below

							*!!*
!!{filter}		Filter [count] lines through the external program
			program {filter}. If {filter} is empty, use previously
			{filter} program. {filter} is initially set to 'shell'
			by default on startup.

							*v_!*
{Visual}!{filter}	Filter the highlighted lines through the external
			program {filter} (for {Visual} see |Visual-mode|). If
			{filter} is empty, use previously {filter} program.
			{filter} is initially set to 'shell' by default on
			startup.

:{range}![!]{filter} [!][arg]				*:range!*
			Filter {range} lines through the external program

> 			{filter}. If no {filter} program is given, the latter

Reviewer: ```suggestion
			{filter} (or 'shell' by default).
```

							*!*

> !{motion}{filter}	Filter {motion} text lines through the external

Reviewer: brackets indicate optional args, in the docs.
```suggestion
!{motion}[filter]	Filter {motion} text lines through the external
```

Author: Thank you for this correction.

> 			program {filter}. If {filter} is empty, use previous
> 			{filter} program, or 'shell' by default.

Reviewer: ```suggestion
			program [filter]. If [filter] is empty, use previous
			[filter] program, or 'shell' by default.
```
